src/fu-util.c

```diff
@@ -2158,6 +2158,26 @@ fu_util_private_free (FuUtilPrivate *priv)
 	g_free (priv);
 }
 
+static gboolean
+fu_util_check_daemon_version (FuUtilPrivate *priv, GError **error)
+{
+	g_autofree gchar *client = g_strdup_printf ("%i.%i.%i",
+						    FWUPD_MAJOR_VERSION,
+						    FWUPD_MINOR_VERSION,
+						    FWUPD_MICRO_VERSION);
+	const gchar *daemon = fwupd_client_get_daemon_version (priv->client);
+
+	if (fu_common_vercmp (daemon, client) < 0) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     _("Unsupported daemon, requires at least %s"), client);
```

Comment:
Maybe make it a strcmp and something like `Unsupported daemon version %s, client version is %s` -- with the binary watching we added recently the daemon and client versions should match exactly, no?

Reply:
changed